src/sql/src/query_model/hir/qgm_from_hir.rs

```diff
@@ -185,7 +182,32 @@ impl FromHir {
                 }
                 Ok(box_id)
             }
-            // HirRelationExpr::CallTable { func, exprs } => todo!(),
+            HirRelationExpr::CallTable { func, exprs } => {
+                // mark the output type of the called function for later
+                let output_type = func.output_type();
+
+                // create box with empty function call argument expressions
+                let call_table = CallTable::new(func, vec![], output_type.keys);
+                let box_id = self.model.make_box(call_table.into());
+
+                // fix function the call arguments (it is a bit awkward that this needs
+                // to be adapted only after creating the box)
+                let exprs = self.generate_exprs(exprs, box_id)?;
+                let mut r#box = self.model.get_mut_box(box_id);
+                if let BoxType::CallTable(call_table) = &mut r#box.box_type {
+                    call_table.exprs = exprs;
+                }
```

Comment:
I think the way to avoid this awkwardness is to change the signature of `make_box` so that instead of taking a `BoxType`, it takes a function `fn(box_id) -> BoxType`.

Reply:
Ingenious! I'll take a note. I also wanted to make the BoxType generaic `T: Into<BoxType>` in order to push the `.into()` calls inside the `make_box` constructor.